src/lsptoolshost/razor/razorEndpoints.ts

```diff
@@ -24,51 +24,94 @@ import { DocumentColorHandler } from '../../razor/src/documentColor/documentColo
-    languageServer.registerOnNotificationWithParams(logNotificationType, (params) =>
+    roslynLanguageServer.registerOnNotificationWithParams(logNotificationType, (params) =>
         razorLogger.log(params.message, params.type)
     );
 
-    if (!razorOptions.cohostingEnabled) {
-        return;
+    if (razorOptions.cohostingEnabled) {
+        registerCohostingEndpoints();
+    } else {
+        registerNonCohostingEndpoints();
     }
 
-    const documentManager = new HtmlDocumentManager(platformInfo, razorLogger);
-    context.subscriptions.push(documentManager.register());
+    return;
 
-    registerRequestHandler<HtmlUpdateParameters, void>('razor/updateHtml', async (params) => {
-        const uri = UriConverter.deserialize(params.textDocument.uri);
-        await documentManager.updateDocumentText(uri, params.text);
-    });
+    //
+    // Local Functions
+    //
+    function registerCohostingEndpoints() {
+        const documentManager = new HtmlDocumentManager(platformInfo, razorLogger);
+        context.subscriptions.push(documentManager.register());
 
-    registerRequestHandler<DocumentColorParams, ColorInformation[]>(DocumentColorRequest.method, async (params) => {
-        const uri = UriConverter.deserialize(params.textDocument.uri);
-        const document = await documentManager.getDocument(uri);
-
-        return await DocumentColorHandler.doDocumentColorRequest(document.uri);
-    });
+        registerRequestHandler<HtmlUpdateParameters, void>('razor/updateHtml', async (params) => {
+            const uri = UriConverter.deserialize(params.textDocument.uri);
+            await documentManager.updateDocumentText(uri, params.text);
+        });
 
-    registerRequestHandler<ColorPresentationParams, ColorPresentation[]>(
-        ColorPresentationRequest.method,
-        async (params) => {
+        registerRequestHandler<DocumentColorParams, ColorInformation[]>(DocumentColorRequest.method, async (params) => {
             const uri = UriConverter.deserialize(params.textDocument.uri);
             const document = await documentManager.getDocument(uri);
 
-            return await ColorPresentationHandler.doColorPresentationRequest(document.uri, params);
-        }
-    );
+            return await DocumentColorHandler.doDocumentColorRequest(document.uri);
+        });
+
+        registerRequestHandler<ColorPresentationParams, ColorPresentation[]>(
+            ColorPresentationRequest.method,
+            async (params) => {
+                const uri = UriConverter.deserialize(params.textDocument.uri);
+                const document = await documentManager.getDocument(uri);
+
+                return await ColorPresentationHandler.doColorPresentationRequest(document.uri, params);
+            }
+        );
+    }
+
+    function registerNonCohostingEndpoints() {
+        registerRequestHandler<ProvideDynamicFileParams, ProvideDynamicFileResponse>(
+            'razor/provideDynamicFileInfo',
```

Comment:
Is it reasonable to use the constants from https://github.com/dotnet/vscode-csharp/blob/d3e1c708488f48c2be988c5913da7da8a5e0f82d/src/lsptoolshost/server/roslynLanguageServer.ts#L88?